ts/Extensions/Boost/WGLRenderer.ts

```diff
@@ -773,9 +773,18 @@ class WGLRenderer {
 
             // Handle the point.color option (#5999)
             const pointOptions = rawData && rawData[i];
-            if (!useRaw && isObject(pointOptions, true)) {
-                if (pointOptions.color) {
-                    pcolor = color(pointOptions.color).rgba as any;
+            if (!useRaw) {
+                if (isObject(pointOptions, true) && pointOptions.color) {
+                    pcolor = color(pointOptions.color).rgba;
+                }
+                if (
+                    Array.isArray(pointOptions) &&
+                    typeof pointOptions[2] === 'string'
```

Comment:
We can't assume that the color is at position 2. If `keys` are not defined, properties are resolved as per the `pointArrayMap` property, which is `[x, y]` for the scatter series. If `keys` are defined, it overrides the pointArrayMap, and the order of items in the data array should follow that of the keys.

If we disable the Boost module, we see how it should work: https://jsfiddle.net/highcharts/7u41mkyc/ . Properties can be given in any position within the keys (and matching data). This setup fails with the PR, breaking the spec for `keys`.

PS: It also fails for columns, indicating that the support for keys is not only failing in the special scatter logic.